service/context.go

```diff
@@ -206,6 +206,8 @@ func (c *response) Flush() {
 }
 
 // CloseNotify is a disguise of http.response.CloseNotify().
+//
+// Deprecated: use `http.Request.Context.Done()` as instead.
 func (c *response) CloseNotify() <-chan bool {
```

Comment:
The implementation is not correct. This method makes users have no way to check if the response supports http.CloseNotifier. Do you have time to fix this in a new PR?

Reply:
So I deleted it directly in this PR? As mentioned in #288

Reply:
I mean that the response may not implement `http.CloseNotifier`.

The method should be separated into two parts:
```
type response struct {...}


type responseWithCloseNotify struct {
    *response
}

func (c *responseWithCloseNotify) CloseNotify() <-chan bool {...}
```

Then wrap response with the new struct.

```

rw = &response{}

if _, ok:=rw.(http.CloseNotifier); ok {
    rw = &responseWithCloseNotify{rw}
}
```

If the method is removed from the go library in the future, the `response` still works.

Reply:
You mean still keep this function? As you replied `Just do it.` in #288 I thought we can just delete this function.

Reply:
It seems that we need to retain it because of the behavior of go library.

Reply:
Fine, I will fix this in a new PR